skglm/experimental/sqrt_lasso.py

```diff
@@ -0,0 +1,230 @@
+import warnings
+import numpy as np
+from numpy.linalg import norm
+from sklearn.exceptions import ConvergenceWarning
+from sklearn.linear_model._base import LinearModel, RegressorMixin
+
+from skglm.penalties import L1
+from skglm.utils import compiled_clone, ST_vec, proj_L2ball
+from skglm.datafits.base import BaseDatafit
+from skglm.solvers.prox_newton import ProxNewton
+
+
+class SqrtQuadratic(BaseDatafit):
+    """Square root quadratic datafit.
+
+    The datafit reads::
+        ||y - Xw||_2 / sqrt(n_samples)
+    """
+
+    def __init__(self):
+        pass
+
+    def get_spec(self):
+        spec = ()
+        return spec
+
+    def params_to_dict(self):
+        return dict()
+
+    def value(self, y, w, Xw):
+        return np.linalg.norm(y - Xw) / np.sqrt(len(y))
+
+    def raw_grad(self, y, Xw):
+        """Compute gradient of datafit w.r.t ``Xw``.
+
+        Raises
+        ------
+            Exception
+                if value of residuals is too small (less than ``1e-10``).
+        """
+        minus_residual = Xw - y
+        norm_residuals = norm(minus_residual)
+
+        if norm_residuals < 1e-2 * norm(y):
+            raise ValueError("SmallResidualException")
+
+        return minus_residual / (norm_residuals * np.sqrt(len(y)))
+
+    def raw_hessian(self, y, Xw):
+        """Upper bound of the hessian w.r.t eigenvalues."""
```

Comment:
Nitpick: I would try to be even more explicit that the upper bound is in the matrix sense, and that this is a diagonal matrix

Reply:
I agree. I have raised this comment already.